tests/Application/Matching/AdapterTest.php

```diff
@@ -111,39 +115,43 @@ public function testItBailsOutAndReleasesFundsIfRetryingDoesntWorkDueToConcurren
-            return $this->storage->decrBy($key, 30_00);
+            return $this->storage->decrBy($key, 40_00);
         });
 
-            $funding = new CampaignFunding();
-            $funding->setId(53);
-            $funding->setAmountAvailable('50');
-            $amountToSubtract = "30";
-
-            $this->sut->subtractAmountWithoutSavingToDB($funding, $amountToSubtract);
-
-            $this->expectException(TerminalLockException::class);
-            // todo - work out where the -100_00 figure here comes from. Message below is just pasted in from
-            // see ticket MAT-332
-            // result of running the test.
-
-         // We initially subtract £30, leaving £20 in fund.
-         // Then we try to subtract another £30, leaving £-10 in fund.
-         //
-        // Each of the 5 auto retries does the same thing, removing another -10 from the fund. Combined with the 30 removed by the other process that leaves
-        // total of 6*10 + 30 = 100 to release.
-
-        // As a test I changed the retry limit in Adapter::$maxPartialAllocateTries from 5 to 500.
-        // That means it has to release -14950_00 at the end. Actually still sort of confused.
-
+        $funding = new CampaignFunding();
+        $funding->setId(53);
+        $funding->setAmountAvailable('50');
+        $amountToSubtract = "30";
 
+        $this->sut->subtractAmountWithoutSavingToDB($funding, $amountToSubtract);
 
-            $this->expectExceptionMessage("Fund 53 balance sub-zero after 6 attempts. Releasing final -10000 'cents'");
+        try {
             $this->sut->subtractAmountWithoutSavingToDB($funding, $amountToSubtract);
+            $this->fail("should have thrown exception on attempt to allocate more than available");
+        } catch (TerminalLockException $exception) {
+            // this -140_00 number is not very important - we already released part of the funds earlier,
+            // so this is just whatever happened to be left to release at the end. It's influenced by what happened
+            // in the callback (simulated other process) because that influences how much we tried to release earlier,
+            // and so this has to balance it out. If we released more earlier in the attempt to get the fund to zero
+            // we would release less here and vice versa.
+            //
+            $this->assertSame(
+                "Fund 53 balance sub-zero after 6 attempts. Releasing final -14000 'cents'",
+                $exception->getMessage()
+            );
+        }
+
+        // fund started with 50, we successfully removed 30, and the callback removed 40 six times. We cleared up after
+        // our unsuccessful attempts to withdraw more, not cleared up after the callback as would be the responsibility
+        // of the other process.
+        \assert(50 - 30 - 40 * 6 === -220);
+        $this->assertSame('-220.00', $this->sut->getAmountAvailable($funding));
```

Comment:
the most important thing is not the individual reservations and releases, but that it ads up to make the total correct at the end. Hence adding this assertion.